src/ReactiveUI/ReactiveList.cs

```diff
@@ -29,19 +29,31 @@ public class ReactiveList<T> : IReactiveList<T>, IReadOnlyReactiveList<T>, IList
 
         protected virtual void raiseCollectionChanging(NotifyCollectionChangedEventArgs args)
         {
-            var handler = this.CollectionChanging;
-            if (handler != null) {
-                handler(this, args);
-            }
+            // WPF doesn't seem to care much about Range notifications in this case, 
+            // so we'll just pass those right through
+            this.CollectionChanging?.Invoke(this, args);
         }
 
         public event NotifyCollectionChangedEventHandler CollectionChanged;
 
         protected virtual void raiseCollectionChanged(NotifyCollectionChangedEventArgs args)
         {
-            var handler = this.CollectionChanged;
-            if (handler != null) {
-                handler(this, args);
+            // WPF throws "Range actions are not supported" under
+            // certain circumstances, so we catch those here first
+            this.CollectionChanged?.Invoke(this, replaceIfUnsupported(args));
+        }
+        
+        NotifyCollectionChangedEventArgs replaceIfUnsupported(NotifyCollectionChangedEventArgs args)
+        {
+            // see System.Windows.Data.ListCollectionView.ValidateCollectionChangedEventArgs
+            switch (args.Action) {
+                case NotifyCollectionChangedAction.Add when args.NewItems.Count != 1:
+                case NotifyCollectionChangedAction.Remove when args.OldItems.Count != 1:
```

Comment:
Do we want to do this for all platforms?

Reply:
I don't think so. WPF/.NET Framework seems to be the only target platform that is picky about these conditions ([src](https://referencesource.microsoft.com/#PresentationFramework/src/Framework/System/Windows/Data/ListCollectionView.cs,0c35e5d5d6bb7c2f)). But I assume that, since this piece of code is under a `#if NET_461` guard, it will get included only there (at least that was my thinking a year ago).

Reply:
Makes sense.